application/client.core/src/app/environment/services/standalone/service.contextmenu.ts

```diff
@@ -17,14 +17,30 @@ export interface IMenu {
     y: number;
 }
 
+export enum EEventType {
+    keydown = 'keydown',
+    mousedown = 'mousedown'
+}
+
+type TEvent = MouseEvent | KeyboardEvent;
+type TClose = () => void | undefined;
+
 export class ContextMenuService {
 
+    private _close: TClose;
     private _subjects: {
         onShow: Subject<IMenu>,
     } = {
         onShow: new Subject<IMenu>(),
     };
 
+    public get close(): TClose {
+        return this._close;
+    }
+
+    public set close(func: TClose) {
+        this._close = func;
+    }
```

Comment:
It isn't a mistake or similar. I mean solution works. But... This is not clean solution and here are a few reasons:
- why close-handler defined outside?
- "who" can define close-handler? Anybody?
- why it's public? (related to previous)

As for me, it would be simpler and in logic of existing code to use `subject`. Why not? Just add new subject/event (in addition to "onShow"), like "onClose" and what's all. Any consumers could subscribe on "onClose" event and do, what they want to do without any risk.